rllib/offline/offline_data.py

```diff
@@ -0,0 +1,152 @@
+import functools
+import logging
+import numpy as np
+from pathlib import Path
+import ray
+from typing import Any, Dict, List, Union
+
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.core.columns import Columns
+from ray.rllib.env.single_agent_episode import SingleAgentEpisode
+from ray.rllib.utils.compression import unpack_if_needed
+from ray.rllib.utils.typing import EpisodeType
+
+logger = logging.getLogger(__name__)
+
+# TODO (simon): Implement schema mapping for users, i.e. user define
+# which row name to map to which default schema name below.
+SCHEMA = [
+    Columns.EPS_ID,
+    Columns.AGENT_ID,
+    Columns.MODULE_ID,
+    Columns.OBS,
+    Columns.ACTIONS,
+    Columns.REWARDS,
+    Columns.INFOS,
+    Columns.NEXT_OBS,
+    Columns.TERMINATEDS,
+    Columns.TRUNCATEDS,
+    Columns.T,
+    # TODO (simon): Add remove as soon as we are new stack only.
+    "agent_index",
+    "dones",
+    "unroll_id",
+]
+
+
+class OfflineData:
```

Comment:
Should we rename this `OfflineDataGroup`? Analogous to `EnvRunnerGroup` and `LearnerGroup`.

Reply:
Mmmmh I am not sure about this. As we have no data group, its simply a single Dataset.

Reply:
Ok, then no.

I see, but inside the Dataset, there could be n sharded actors, correct?

Reply:
As far as I understand Ray Data a `data.Dataset` holds multiple blocks (shards)  of ArrowTables, but these are not held by actors but instead are put in PyArrow format into object store and therefore accessible by all actors.